aspnetcore/fundamentals/configuration/options.md

```diff
@@ -0,0 +1,256 @@
+
+[!code-csharp[Main](options/sample/Startup.cs?name=snippet_Example1)]
+
+The following page model uses [constructor dependency injection](xref:fundamentals/dependency-injection#what-is-dependency-injection) with [IOptions&lt;TOptions&gt;](/dotnet/api/Microsoft.Extensions.Options.IOptions-1) to access the settings (*Pages/Index.cshtml.cs*):
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?range=9)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet2&highlight=2,8)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet_Example1)]
+
+The sample's *appsettings.json* file specifies values for `option1` and `option2`:
+
+[!code-json[Main](options/sample/appsettings.json)]
+
+When the app is run, the page model's `OnGet` method returns a string showing the option class values:
+
+```html
+option1 = value1_from_json, option2 = -1
+```
+
+## Configure simple options with a delegate
+
+Configuring simple options with a delegate is demonstrated as Example &num;2 in the [sample app](https://github.com/aspnet/docs/tree/master/aspnetcore/fundamentals/configuration/options/sample).
+
+Use a delegate to set options values. The sample app uses the `MyOptionsWithDelegateConfig` class (*Models/MyOptionsWithDelegateConfig.cs*):
+
+[!code-csharp[Main](options/sample/Models/MyOptionsWithDelegateConfig.cs?name=snippet1)]
+
+In the following code, a second `IConfigureOptions<TOptions>` service is added to the service container. It uses a delegate to configure the binding with `MyOptionsWithDelegateConfig` (*Startup.cs*):
+
+[!code-csharp[Main](options/sample/Startup.cs?name=snippet_Example2)]
+
+*Index.cshtml.cs*:
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?range=10)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet2&highlight=3,9)]
+
+[!code-csharp[Main](options/sample/Pages/Index.cshtml.cs?name=snippet_Example2)]
+
+You can add multiple configuration providers. Configuration providers are available in NuGet packages. They're applied in order that they're registered.
+
+Each call to [Configure&lt;TOptions&gt;](/dotnet/api/microsoft.extensions.options.iconfigureoptions-1.configure) adds an `IConfigureOptions<TOptions>` service to the service container. In the preceding example, the values of `Option1` and `Option2` are both specified in *appsettings.json*, but the values of `Option1` and `Option2` are overridden by the configured delegate.
+
+When more than one configuration service is enabled, the last configuration source specified *wins* and sets the configuration value. When the app is run, the page model's `OnGet` method returns a string showing the option class values:
+
+```html
+delegate_option1 = value1_configured_by_delgate, delegate_option2 = 500
+```
+
+## Sub-options configuration
```

Comment:
According to the Chicago Manual of Style, there should be no hyphenation after the prefix sub in words such as sub-heading.